crates/storage_impl/src/payments/payment_intent.rs

```diff
@@ -1086,6 +1124,302 @@ impl<T: DatabaseStore> PaymentIntentInterface for crate::RouterStore<T> {
             .await
     }
 
+    #[cfg(all(feature = "v2", feature = "olap"))]
+    #[instrument(skip_all)]
+    async fn get_filtered_payment_intents_attempt(
+        &self,
+        state: &KeyManagerState,
+        merchant_id: &common_utils::id_type::MerchantId,
+        constraints: &PaymentIntentFetchConstraints,
+        merchant_key_store: &MerchantKeyStore,
+        storage_scheme: MerchantStorageScheme,
+    ) -> error_stack::Result<Vec<(PaymentIntent, Option<PaymentAttempt>)>, StorageError> {
+        use diesel::NullableExpressionMethods as _;
+        use futures::{future::try_join_all, FutureExt};
+
+        use crate::DataModelExt;
+
+        let conn = connection::pg_connection_read(self).await.switch()?;
+        let conn = async_bb8_diesel::Connection::as_async_conn(&conn);
+        let mut query = DieselPaymentIntent::table()
+            .filter(pi_dsl::merchant_id.eq(merchant_id.to_owned()))
+            .left_join(
+                payment_attempt_schema::table
+                    .on(pi_dsl::active_attempt_id.eq(pa_dsl::id.nullable())),
+            )
+            .into_boxed(); // Filtering on merchant_id for payment_attempt is not required for v2
```

Comment:
Can you also include the reason for not filtering payment attempts based on merchant ID?